tests/ctl/api/test_seed.py

```diff
@@ -436,3 +444,193 @@ async def test_load_orginizations(loguru_caplog, async_session, monkeypatch):
+            assert resources_dict
+        except Exception as exc:
+            assert not exc, error_message
+
+    @patch.dict(os.environ, SAMPLE_ENV_VARS, clear=True)
+    async def test_load_sample_connections(self):
+        """
+        Ensure that the sample connections file in the sample project can be
+        parsed and loaded by the load_sample_connections_from_project() function.
+        This makes sure we don't make some changes to the sample project files
+        that aren't automatically loaded into the database via this function.
+
+        NOTE: If you've found this test, that probably means you were making
+        some changes to the code and this failed unexpectedly. Maybe you removed
+        a field, changed a default, or wanted to edit some sample data?
+
+        To fix the test, you just need to ensure the code has all the logic it
+        needs to parse everything from this directory:
+        - src/fides/data/sample_project/sample_connections/*.yml
+
+        See src/fides/api/ctl/database/samples.py for details.
+
+        Sorry for the trouble, but we want to ensure there isn't a subtle bug
+        sneaking into our sample project code!
+        """
+        error_message = (
+            "Unexpected error loading sample connections; did you make changes to the sample project? "
+            "See tests/ctl/api/test_seed.py for details."
+        )
+        connections = []
+        try:
+            connections = samples.load_sample_connections_from_project()
+        except Exception as exc:
+            assert not exc, error_message
+
+        # Assert that only the connections with all their secrets are returned
+        assert len(connections) == 2
+        assert sorted([e.key for e in connections]) == [
+            "postgres_connector",
+            "stripe_connector",
+        ]
+
+        # Assert that variable expansion worked as expected
+        postgres = [e for e in connections if e.connection_type == "postgres"][0].dict()
+        assert postgres["secrets"]["host"] == "test-var-expansion"
+        assert postgres["secrets"]["port"] == 9090
+
+    @patch.dict(
+        os.environ,
+        {
+            "TEST_VAR_1": "var-1",
+            "TEST_VAR_2": "var-2",
+        },
+        clear=True,
+    )
+    async def test_load_sample_yaml_file(self):
+        """
+        Test that we can safely load, parse, and perform variable expansion on
+        a sample project file.
+        """
+        sample_str = dedent(
+            """\
+            connection:
+              - key: test_connection
+                name: Test Connector $TEST_VAR_1
+                connection_type: postgres
+                access: write
+                secrets:
+                  host: test-host
+                  port: 9001
+                  dbname: $TEST_VAR_2
+                  username: user-${TEST_VAR_2}
+                  password: ${TEST_VAR_1}-${TEST_VAR_2}
```

Comment:
This is a good example of what `expandvars()` supports. It's actually more fully featured and allows for bash-isms like default values (`${FOO:-defaultvaluehere}`)...